docs/06-contributors/999-rfcs/000-stories/story-15.md

```diff
@@ -0,0 +1,173 @@
+# User Story 15 - Task List with Api Gateway and Redis
+
+> **Status**: Draft, Expected released on 2023/03/16
+
+
+The following code is an inital implementation of TaskList with api gateway and a redis db 
+
+## New landuage and SDK features it introduces
+
+- [x] bring untyped
+  - [x] bring external npm package (axios)
+  - [x] bring an internal nodejs stdlib (RegEx)
+  - [x] How does untyped works with numeric operations?
+  - [x] Do we cast untyped? 
+- [x] enum & duration that can be included inside json
+- [x] It leverages setting explicit permissions (using the `this.inflight` API, described [here](https://github.com/winglang/wing/pull/1610))
+- [x] Using interface 
+- [x] Using await when calling a untyped function that returns a promise
+- [ ] bring cdktf
+- [ ] use redis instead of bucket
+- [ ] code that updates estimation and duration from REST post command
+
+## Discussion topics
+- `get` vs `on_get` - Should the `cloud.api` API have the `on_` prefix to match `cloud.bucket` API and also to allow calling
+the api get/post/delete/put commands (`api.get(url)` vs `api.on_get(path, inflight ())`
+- Routing params - I have used express's synatx for dynamic parts of the path for the api gateway `get("/task/:id"`, 
+but then I noticed that [aws](https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-create-api-step-by-step.html) 
+uses the `{id}` syntax. What should be the the right syntax?
+- Bring internal nodejs types - how do we bring somethink that we can't require, like `RegEx`
+- From anything to Json - can I cast something from bring untyped to Json? 
+- Calling await 
+
+## Code 
+```ts (wing)
+
+bring cloud;
+
+// TODO discuss how we bring untyped something like RegEx from JavaScript 
```

Comment:
How about this:

```js
bring untyped js;

// and then
new js.RegExp(...)
js.Math.max()
```

Reply:
love it

Reply:
Done

Reply:
`bring untyped js` seems too wide. Can we limit the scope somehow? `bring untyped js.RegExp as RegExp`?

Reply:
> `bring untyped js` seems too wide. Can we limit the scope somehow? `bring untyped js.RegExp as RegExp`?

Here are my thoughts about this: 

I think that the difference here is about the maturity of wing.

If you use wing today, `bring untyped js;` will be more empathic to your needs, because, at beta, we are going to see the bring untyped quite often in wing code. Once wing is closer to GA, we expect the developer to use `bring untyped` less common, in that case the `bring untyped js.RegExp as RegExp;` is more empathic because it shows in a glance all the untyped imports this code is bringing.

@eladb , this is a question of language UX

Reply:
This is resolved as part of the last commit